api/v1alpha1/argocd_types.go

```diff
@@ -642,6 +642,13 @@ type ArgoCDStatus struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Dex",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
 	Dex string `json:"dex,omitempty"`
 
+	// SSOConfig is an indicator of misconfiguration where multiple SSO providers are configured
```

Comment:
The comment message is confusing. Does this sound good ?
"SSOConfig defines the status of SSO configuration"

Reply:
Description for other status fields all start with"xxx is a simple, high-level summary of where the Argo CD ...", I tried to keep it consistent but couldn't find a better way to put it. "SSOConfig defines the status of SSO configuration" I think this way user might be confused like what does "status of SSO configuration" mean.

@wtam2018 @jannfis Any idea of how to put the description of this new status field?

Reply:
I think @iam-veeramalla looks good.

Reply:
Just out of curiosity, we have only defined the meanings of the different states (success/failed etc) of `ssoConfig` internally in the code, right?

How would an average user know about this?

Reply:
@jaideepr97 The generated manifest file has the description of definition of each state, I don't know how they can check it out without look into the source code though, if this is what you mean. I'm also adding comments to the `reconcileStatusSSOConfig` function for developer's reference.

Reply:
Yeah, I mean the average user is not going to look at code, they'll just see `ssoConfig: success/failure` in their CR 

Maybe we should document this somewhere in our release notes then? @wtam2018 @iam-veeramalla

Reply:
I agree that as a user if I read `status.ssoConfig: Failed` it's hard to tell what exactly went wrong... It could be I put wrong value for the sso configuration, or I missed the sso configuration with no detail explained.

Reply:
The error message with full details goes to the log, right?  We can also generate an event.

Reply:
Thanks for the review, let's add the explanation to the release note.

Reply:
@wtam2018 Yes, there is an error log says "Installation of multiple SSO providers is not permitted. Please choose a single provider for Argo CD %s in namespace %s."